internal/ethapi/celo_api_test.go

```diff
@@ -115,63 +77,457 @@ func TestNewRPCTransactionLegacy(t *testing.T) {
+			// CeloDynamicFeeTxs are deprecated after cel2 so we need to ensure cel2time is not activated
+			config := allEnabledChainConfig()
+			cel2Time := uint64(1000)
+			config.Cel2Time = &cel2Time
+			s := types.MakeSigner(config, new(big.Int).SetUint64(blockNumber), blockTime)
+
+			tx := types.NewTx(&types.CeloDynamicFeeTx{
+				ChainID:             config.ChainID,
+				Nonce:               nonce,
+				Gas:                 gasLimit,
+				GasFeeCap:           feeCap,
+				GasTipCap:           tipCap,
+				FeeCurrency:         &feeCurrency,
+				GatewayFee:          gatewayFee,
+				GatewayFeeRecipient: &gatewayFeeRecipient,
+
+				To:    &to,
+				Value: value,
+				Data:  []byte{},
+			})
+
+			signed, err := types.SignTx(tx, s, key)
+			require.NoError(t, err)
+
+			rpcTx := newRPCTransaction(signed, blockhash, blockNumber, blockTime, transactionIndex, baseFee, config, nil)
+			checkTxFields(t, signed, rpcTx, s, blockhash, blockNumber, transactionIndex, overrides)
+		})
+
+		t.Run("CeloDynamicFeeTxV2", func(t *testing.T) {
+			tx := types.NewTx(&types.CeloDynamicFeeTxV2{
+				ChainID:     config.ChainID,
+				Nonce:       nonce,
+				Gas:         gasLimit,
+				GasFeeCap:   feeCap,
+				GasTipCap:   tipCap,
+				FeeCurrency: &feeCurrency,
+
+				To:    &to,
+				Value: value,
+				Data:  []byte{},
+			})
+
+			signed, err := types.SignTx(tx, s, key)
+			require.NoError(t, err)
+
+			rpcTx := newRPCTransaction(signed, blockhash, blockNumber, blockTime, transactionIndex, baseFee, config, nil)
+			checkTxFields(t, signed, rpcTx, s, blockhash, blockNumber, transactionIndex, overrides)
+		})
+	})
+
+	t.Run("PostCel2MinedDynamicTxs", func(t *testing.T) {
+		receipt := &types.Receipt{}
+		receipt.EffectiveGasPrice = big.NewInt(1234)
+		effectiveGasPrice := func(t *testing.T, tx *types.Transaction, rpcTx *RPCTransaction) {
+			assert.Equal(t, (*hexutil.Big)(receipt.EffectiveGasPrice), rpcTx.GasPrice)
+		}
+		overrides := map[string]func(*testing.T, *types.Transaction, *RPCTransaction){"gasPrice": effectiveGasPrice}
+
+		config := allEnabledChainConfig()
+		s := types.MakeSigner(config, new(big.Int).SetUint64(blockNumber), blockTime)
+
+		t.Run("CeloDynamicFeeTxV2", func(t *testing.T) {
+			// For a pre gingerbread mined dynamic fee tx we expect the gas price to be unset.
+			tx := types.NewTx(&types.CeloDynamicFeeTxV2{
+				ChainID:     config.ChainID,
+				Nonce:       nonce,
+				Gas:         gasLimit,
+				GasFeeCap:   feeCap,
+				GasTipCap:   tipCap,
+				FeeCurrency: &feeCurrency,
+
+				To:    &to,
+				Value: value,
+				Data:  []byte{},
+			})
+
+			signed, err := types.SignTx(tx, s, key)
+			require.NoError(t, err)
+
+			rpcTx := newRPCTransaction(signed, blockhash, blockNumber, blockTime, transactionIndex, baseFee, config, receipt)
+			checkTxFields(t, signed, rpcTx, s, blockhash, blockNumber, transactionIndex, overrides)
+		})
 	})
 }
 
-func checkRPCTransactionFields(
+func allEnabledChainConfig() *params.ChainConfig {
```

Comment:
Any specific readon why `AllDevChainProtocolChanges` isn't enough?

Reply:
I'm modifying the config in the tests, so I don't want those modifications affecting other tests.